src/ext/java/org/opentripplanner/ext/siri/updater/azure/AbstractAzureSiriUpdater.java

```diff
@@ -186,29 +191,40 @@ public boolean isPrimed() {
     return this.isPrimed;
   }
 
-  public void setPrimed(boolean primed) {
-    isPrimed = primed;
+  private void setPrimed() {
+    isPrimed = true;
   }
 
   @Override
   public String getConfigRef() {
     return this.configRef;
   }
 
-  protected String fetchInitialData(URI uri) {
-    // Maybe put this in the config?
-    HttpHeaders rh = HttpHeaders.of().acceptApplicationXML().build();
-    String initialData;
+  /**
+   * Returns None for empty result
+   */
+  protected Optional<ServiceDelivery> fetchInitialSiriData(URI uri) {
+    var headers = HttpHeaders.of().acceptApplicationXML().build().asMap();
+
     try (OtpHttpClient otpHttpClient = new OtpHttpClient()) {
-      initialData =
-        otpHttpClient.getAndMap(
-          uri,
-          Duration.ofMillis(timeout),
-          rh.asMap(),
-          is -> CharStreams.toString(new InputStreamReader(is))
-        );
+      var t1 = System.currentTimeMillis();
+      var siriOptional = otpHttpClient.executeAndMapOptional(
+        new HttpGet(uri),
+        Duration.ofMillis(timeout),
+        headers,
+        SiriXml::parseXml
+      );
+      var t2 = System.currentTimeMillis();
+      LOG.info("Fetched initial data in {} ms", (t2 - t1));
+
+      if (siriOptional.isEmpty()) {
+        LOG.info("Got status 204 'No Content', handling gracefully.");
+        return Optional.empty();
+      }
+
+      var serviceDelivery = siriOptional.get().getServiceDelivery();
+      return Optional.ofNullable(serviceDelivery);
```

Comment:
Not a huge deal but I like to write the optional handling in this style.

```suggestion
     return siriOptional.or(() -> {
        LOG.info("Got status 204 'No Content', handling gracefully.");
        return Optional.empty();
      }).map(Siri::getServiceDelivery);     
```

Reply:
For some traditional Java developers this style is a bit strange so I leave it up to you.

Reply:
Yeah, java optional handling has a tendency to get a bit clunky. But personally I'm not a fan of replacing if-conditionals with chains of Optionals, since it obscures the conditional. Also I wanted to get rid of the potential null from getServiceDelivery.

I got a bit inspired and cleaned this up a little bit though :)